### PR TITLE
Make message informative for a missing read group in the covariates table

### PR DESCRIPTION
**Pull request by ronlevine**
Implements #2430.
Made the exception message informative.
Created a test where  the input covariates table file has a read group in the quality score (RecalTable1) or covariates (RecalTable2) report that is not in read group report (RecalTable0).

### CODE REVIEW

**Comment by ronlevine** [comment]
@vdauwera Please review.

**Comment by vdauwera** [comment]
Do you plan to address the question of empty recal files separately?

**Comment by ronlevine** [comment]
I can add that to this PR.

**Comment by vdauwera** [comment]
Splendid, thank you

**Comment by ronlevine** [comment]
@vdauwera Added the informative message for an empty recal file.

File: src/main/java/org/broadinstitute/hellbender/utils/report/GATKReport.java [hunk 1]
```diff
@@ -93,9 +94,12 @@ private void loadReport(Reader in) {
         try {
             reportHeader = reader.readLine();
         } catch (IOException e) {
-            throw new UserException("Couldn't read RecalibrationReport", e);
+            throw new UserException("Could not read the " + RECAL_FILE, e);
```
Comment (vdauwera) [inline src/main/java/org/broadinstitute/hellbender/utils/report/GATKReport.java]: Both here and below I would either leave out "the" or use "the file" instead.

**Comment by vdauwera** [comment]
One tiny comment, otherwise good to go!

**Pull request merged**